heron/schedulers/src/java/org/apache/heron/scheduler/kubernetes/KubernetesContext.java

```diff
@@ -291,31 +309,202 @@ public static boolean getPersistentVolumeClaimDisabled(Config config) {
           volumes.put(volumeName, volume);
         }
 
-        /* Validate Claim and Storage Class names.
-          [1] `claimNameNotOnDemand`: checks for a `claimName` which is not `OnDemand`.
-          [2] `storageClassName`: Check if it is the provided `option`.
-          Conditions [1] OR [2] are True, then...
-          [3] Check for a valid lowercase RFC-1123 pattern.
-         */
-        boolean claimNameNotOnDemand =
-            KubernetesConstants.VolumeClaimTemplateConfigKeys.claimName.equals(key)
-                && !KubernetesConstants.LABEL_ON_DEMAND.equalsIgnoreCase(value);
-        if ((claimNameNotOnDemand // [1]
-            ||
-            KubernetesConstants.VolumeClaimTemplateConfigKeys.storageClassName.equals(key)) // [2]
-            && !matcher.reset(value).matches()) { // [3]
-          throw new TopologySubmissionException(
-              String.format("Option `%s` value `%s` does not match lowercase RFC-1123 pattern",
-                  key, value));
-        }
-
         volume.put(key, value);
       }
     } catch (IndexOutOfBoundsException | IllegalArgumentException e) {
-      final String message = "Invalid Persistent Volume Claim CLI parameter provided";
+      final String message = "Invalid Volume configuration option provided on CLI";
       LOG.log(Level.CONFIG, message);
       throw new TopologySubmissionException(message);
     }
+
+    // All Volumes must contain a path.
+    for (Map.Entry<String, Map<KubernetesConstants.VolumeConfigKeys, String>> volume
+        : volumes.entrySet()) {
+      final String path = volume.getValue().get(KubernetesConstants.VolumeConfigKeys.path);
+      if (path == null || path.isEmpty()) {
+        throw new TopologySubmissionException(String.format("Volume `%s`: All Volumes require a"
+            + " 'path'.", volume.getKey()));
+      }
+    }
+
+    return volumes;
+  }
+
+  /**
+   * Collects parameters form the <code>CLI</code> and validates options for <code>PVC</code>s.
+   * @param config Contains the configuration options collected from the <code>CLI</code>.
+   * @param isExecutor Flag used to collect CLI commands for the <code>Executor</code> and <code>Manager</code>.
+   * @return A mapping between <code>Volumes</code> and their configuration <code>key-value</code> pairs.
+   * Will return an empty list if there are no Volume Claim Templates to be generated.
+   */
+  public static Map<String, Map<KubernetesConstants.VolumeConfigKeys, String>>
+      getVolumeClaimTemplates(Config config, boolean isExecutor) {
```

Comment:
Would it be worthwhile to mark this parameters passed in the function as `final`?

Reply:
Changes affected.